src/auth_scheme/private_token.ts

```diff
@@ -285,6 +286,123 @@ export class Token {
     }
 }
 
+// enum {
+//     reserved(0),
+//     (65535)
+// } ExtensionType;
+export type ExtensionType = number;
+
+export class Extension {
+    // This class represents the following structure:
+    // See https://www.ietf.org/archive/id/draft-wood-privacypass-auth-scheme-extensions-01.html#section-3-2
+    //
+    // struct {
+    //     ExtensionType extension_type;
+    //     opaque extension_data<0..2^16-1>;
+    // } Extension;
+
+    constructor(
+        public extensionType: ExtensionType,
+        public extensionData: Uint8Array,
+    ) {
+        if (extensionType < 0 || extensionType >= MAX_UINT16) {
```

Comment:
maybe we should also check that it is an integer number
```suggestion
        if (extensionType <= 0 || extensionType >= MAX_UINT16) {
```

Reply:
0 is allowed, even though it's reserved.
I've added `extensionType % 1 !== 0` though, to check it's an integer.